pkg/engine/task/apply_conventions.go

```diff
@@ -25,30 +26,21 @@ import (
 
 const basePath = "/kustomize"
 
-// ExecutionMetadata contains ExecutionMetadata associated with current plan being executed
-type ExecutionMetadata struct {
-	engineMetadata
-
-	planName  string
-	phaseName string
-	stepName  string
-}
-
-// kubernetesObjectEnhancer takes your kubernetes template and kudo related metadata and applies them to all resources in form of labels
+// KubernetesObjectEnhancer takes your kubernetes template and kudo related Metadata and applies them to all resources in form of labels
 // and annotations
 // it also takes care of setting an owner of all the resources to the provided object
-type kubernetesObjectEnhancer interface {
-	applyConventionsToTemplates(templates map[string]string, metadata ExecutionMetadata) ([]runtime.Object, error)
+type KubernetesObjectEnhancer interface {
+	ApplyConventionsToTemplates(templates map[string]string, metadata ExecutionMetadata) ([]runtime.Object, error)
```

Comment:
see for me this is kind of breaking the encapsulation, I think this enhancing is a bit of implementation detail that should be encapsulated in a package and this method should not be public.

Just putting it out there, we don't have to do it in this PR

Reply:
You're right and the only reason it's public is that it is used in the `plan_execution_engine` and its test. I think, the right move would be to move the `plan_execution_engine` into the `\engine` package but that would create even more conflicts in this PR. Wdyt?

Reply:
Yeah sure, I am fine with doing that as a follow-up... :)